### ISSUE TITLE
Support for App Store Promotions

### ISSUE THREAD

**Issue by ekurutepe** [benjaminmayo/merchantkit#20]
This is a documentation question. 

Does MerchantKit support App Store Promotions out of the box? 

I see that `paymentQueue(_:shouldAddStorePayment:for:)` is implemented and it always returns true. Is this enough to opt-in to App Store Promotions or does a consumer app have to configure MerchantKit in some way for App Store Promotions to work? 

Thanks a bunch!

**Comment by benjaminmayo** [comment]
It will ‘work’ in that the In-App Purchase sheet will be shown as soon as the event is processed (i.e tapping the Buy button in the App Store) and the purchase will complete successfully. There’s a URL you can use to test this interaction documented by Apple somewhere. 

In the future, the framework will allow for deferring these payments from starting immediately. For instance, if the user is in the middle of doing something important, the pending purchase can wait until the user’s task is done and the app is ready to continue with it.  This is the equivalent of returning false the `StoreKit` delegate method and holding onto the `SKPayment` until later. `MerchantKit` does not offer that level of control yet.

**Comment by benjaminmayo** [comment]
Closing this as resolved. I have committed to the `master` branch explicit support for the In-App Purchase Promotion, [exposed as an optional method](https://github.com/benjaminmayo/merchantkit/blob/47ab2cff480862f3995db03cb241fae8c8b593ff/Source/MerchantDelegate.swift#L10) on `MerchantDelegate`. The comment above describes the state of all releases to date, and the new feature will be rolled into the next tagged release of `MerchantKit`.
